Windows CI began to fail on 2026-04-07: `msg="fetch failed after status: 404 Not Found" host="localhost:49744"`

https://github.com/moby/moby/actions/runs/24048288313/job/70138697631

```
=== Failed
=== FAIL: client TestIntegration/slice=1-4/TestCacheExportCacheKeyLoop/worker=dockerd-containerd (7.50s)
time="2026-04-06T20:07:00Z" level=info msg="fetch failed after status: 404 Not Found" host="localhost:49744"
    testing.go:1913: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
    --- FAIL: TestIntegration/slice=1-4/TestCacheExportCacheKeyLoop/worker=dockerd-containerd (7.50s)

=== FAIL: client TestIntegration (7.51s)
    run.go:412: 
        	Error Trace:	D:/a/moby/moby/buildkit/util/testutil/integration/run.go:412
        	            				C:/hostedtoolcache/windows/go/1.26.1/x64/src/sync/once.go:78
        	            				C:/hostedtoolcache/windows/go/1.26.1/x64/src/sync/once.go:69
        	            				D:/a/moby/moby/buildkit/util/testutil/integration/run.go:409
        	            				D:/a/moby/moby/buildkit/util/testutil/integration/run.go:239
        	            				C:/hostedtoolcache/windows/go/1.26.1/x64/src/runtime/asm_amd64.s:1771
        	Error:      	Received unexpected error:
        	            	unexpected status from POST request to http://localhost:49744/v2/library/nanoserver/blobs/uploads/: 500 Internal Server Error
        	            	unknown: unknown error
        	            	github.com/moby/buildkit/util/contentutil.copyChain
        	            		D:/a/moby/moby/buildkit/util/contentutil/copy.go:117
        	            	github.com/moby/buildkit/util/contentutil.CopyChain
        	            		D:/a/moby/moby/buildkit/util/contentutil/copy.go:79
        	            	github.com/moby/buildkit/util/testutil/integration.copyImagesLocal
        	            		D:/a/moby/moby/buildkit/util/testutil/integration/run.go:328
        	            	github.com/moby/buildkit/util/testutil/integration.(*Mirror).AddImages
        	            		D:/a/moby/moby/buildkit/util/testutil/integration/run.go:458
        	            	github.com/moby/buildkit/util/testutil/integration.Run.lazyMirrorRunnerFunc.func3.1
        	            		D:/a/moby/moby/buildkit/util/testutil/integration/run.go:412
        	            	sync.(*Once).doSlow
        	            		C:/hostedtoolcache/windows/go/1.26.1/x64/src/sync/once.go:78
        	            	sync.(*Once).Do
        	            		C:/hostedtoolcache/windows/go/1.26.1/x64/src/sync/once.go:69
        	            	github.com/moby/buildkit/util/testutil/integration.Run.lazyMirrorRunnerFunc.func3
        	            		D:/a/moby/moby/buildkit/util/testutil/integration/run.go:409
        	            	github.com/moby/buildkit/util/testutil/integration.Run.Run.func2.func4
        	            		D:/a/moby/moby/buildkit/util/testutil/integration/run.go:239
        	            	testing.tRunner
        	            		C:/hostedtoolcache/windows/go/1.26.1/x64/src/testing/testing.go:2036
        	            	runtime.goexit
        	            		C:/hostedtoolcache/windows/go/1.26.1/x64/src/runtime/asm_amd64.s:1771
        	Test:       	TestIntegration

DONE 2 tests, 2 failures in 7.565s
```

--- a/.github/workflows/buildkit.yml
+++ b/.github/workflows/buildkit.yml
@@ -223,7 +223,9 @@ jobs:
               -v "${{ github.workspace }}\go-build:C:\Users\ContainerAdministrator\AppData\Local\go-build" `
               -v "${{ github.workspace }}\go\pkg\mod:C:\gopath\pkg\mod" `
               ${{ env.TEST_IMAGE_NAME }} hack\make.ps1 -Daemon -Client
-          go install github.com/distribution/distribution/v3/cmd/registry@latest
+          go install "github.com/distribution/distribution/v3/cmd/registry@${{ env.REGISTRY_VERSION }}"
+        env:
+          REGISTRY_VERSION: "v3.0.0"
 
       - name: Checkout BuildKit
         uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2

